sdk/servicebus/service-bus/src/serviceBusMessageBatch.ts

```diff
@@ -123,6 +136,42 @@ export class ServiceBusMessageBatchImpl implements ServiceBusMessageBatch {
     return this._encodedMessages.length;
   }
 
+  /**
+   * Generates an AMQP message that contains the provided encoded messages and annotations.
+   *
+   * @private
+   * @param {Buffer[]} encodedMessages The already encoded messages to include in the AMQP batch.
+   * @param {MessageAnnotations} [annotations] The message annotations to set on the batch.
+   * @param {{ [key: string]: any }} [applicationProperties] The application properties to set on the batch.
+   * @param {{ [key: string]: string }} [messageProperties] The message properties to set on the batch.
+   * @returns {Buffer}
+   * @memberof ServiceBusMessageBatchImpl
+   */
+  private _generateBatch(
+    encodedMessages: Buffer[],
+    annotations?: MessageAnnotations,
+    applicationProperties?: { [key: string]: any },
+    messageProperties?: { [key: string]: string }
+  ): Buffer {
+    const batchEnvelope: AmqpMessage = {
+      body: RheaMessageUtil.data_sections(encodedMessages)
```

Comment:
```suggestion
      body: RheaMessageUtil.data_sections(encodedMessages),
      message_annotations: annotations,
      application_properties: applicationProperties
```

Just wondering if there's any reason we can't just set them all up front?

Reply:
Can do.

Reply:
Done